### DISCUSSION TITLE
[Custom Fixer] How can I rename a file only when --dry-run is disabled in a custom fixer??

### DISCUSSION THREAD

**Discussion by c3zi** [PHP-CS-Fixer/PHP-CS-Fixer#6811] [Q&A] [answered]
## Support question

Is it possible to rename a file in a custom fixer only when `--dry-run` mode is not set? How can I detect when this option is enabled in a custom fixer?

**The scenario for my custom fixer:**
1. Change a class name when there is no `Exception` suffix in exception class (by adding a suffix)
2. Change the file name when above condition is fulfilled 

When `--dry-run` option is not set do not change a file name. 

I am able to use `rename` function in a custom fixer but then even if `--dry-run` option is set the fixer change a file name.
I would like to know whether this scenario is ok for PHP-CS-FIXER or maybe this tool can only change file's content?

**Comment by Wirone** [comment, marked as answer]
I believe this is scenario that currently is not covered, because Fixer's runner does not pass such info to each fixer. With dry run option content is not changed, but potential diff is known. When it comes to file name's change, runner can't prevent it.

But you can workaround it with a pretty dirty way, by using `global $argv;` in `applyFix()`, and checking if `--dry-run` is there 😅 .

**Reply by c3zi** [reply]
@Wirone Thx, so dirty and .. so simple ;-]

**Reply by c3zi** [reply]
Anyway file's name changing is more complicated I thought :) If I simply change the filename there is a problem with an old file path. PHP-CS-Fixer still has link to the old one.
There should be something like `teardown` in every fixer to be run after all fixers finish their job. Then it should be easily possible to change a file name at the end of the fixing process.

**Reply by Wirone** [reply]
@c3zi feel free to create [RFC](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/categories/ideas) which we can discuss 🙂
